test/cpp/api/optim.cpp

```diff
@@ -198,6 +206,10 @@ TEST(OptimTest, XORConvergence_SGD) {
       SGDOptions(0.1).momentum(0.9).nesterov(true).weight_decay(1e-6)));
 }
 
+TEST(OptimTest, XORConvergence_LBFGS) {
+  ASSERT_TRUE(test_optimizer_xor<LBFGS>(LBFGSOptions(1.0).line_search_fn("strong_wolfe")));
```

Comment:
Would `ASSERT_TRUE(test_optimizer_xor<LBFGS>(LBFGSOptions(1.0)));` also work? XD

Reply:
great idea! added the test for optim without line_search_fn :D